spec/models/image_uploader_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+require 'carrierwave/test/matchers'
+require 'pry'
+describe ListingImageUploader, :type => :model do
+  include CarrierWave::Test::Matchers
+
+  before do
+    ListingImageUploader.enable_processing = true
+    @listing = create(:listing)
+    @user = create(:user)
+    @image1 = ListingImage.create(image: File.open(File.join(Rails.root, 'app/assets/images/denver.jpg')))
+    @image2 = ListingImage.create(image: File.open(File.join(Rails.root, 'app/assets/images/seattle.jpg')))
+    @listing << @image1
+    @listing << @image2
+    @listing.save
+  end
+
+  after do
+    MyUploader.enable_processing = false
+    @listing.remove!
+  end
+
+  context 'the file type' do
+    it "should only contain jpg jpeg png or tiff" do
+      ListingImage.create(image: File.open(File.join(Rails.root, 'app/assets/images/denver.pdf'))).should be_false
```

Comment:
Line is too long. [114/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.